src/util/sitemapper.ts

```diff
@@ -191,8 +189,10 @@ export class SitemapReader extends EventEmitter {
         { fullUrl, seedUrl },
         "sitemap",
       );
-      this._parseSitemapFromResponse(fullUrl, resp);
+      await this.parseSitemapFromResponse(fullUrl, resp);
```

Comment:
This was the issue - the parseSitemapFromResponse would start parsing, driven by callbacks, and would not wait for parsing to complete before moving on to next sitemap. Now passing down resolve/reject and awaiting a promise via the events..